api/src/controllers/mission.js

```diff
@@ -114,6 +116,33 @@ router.get("/structure/:structureId", passport.authenticate("referent", { sessio
   }
 });
 
+router.put("/:id/structure/:structureId", passport.authenticate("referent", { session: false }), async (req, res) => {
+  try {
+    const { error: errorId, value: checkedId } = validateId(req.params.id);
+    const { error: errorStructureId, value: checkedStructureId } = validateId(req.params.structureId);
+    if (errorId || errorStructureId) return res.status(400).send({ ok: false, code: ERRORS.INVALID_PARAMS, error });
+    const structure = await StructureObject.findById(checkedStructureId);
+    const mission = await MissionObject.findById(checkedId);
+    if (!mission) return res.status(404).send({ ok: false, code: ERRORS.NOT_FOUND });
+    const missionReferent = await MissionObject.find({ tutorId: mission.tutorId });
+    if (!missionReferent || missionReferent.length > 1) return res.status(405).send({ ok: false, code: ERRORS.OPERATION_NOT_ALLOWED });
```

Comment:
le `!missionReferent` ne matchera pas avec le message qu'on affiche dans le front
il faut faire 2 codes différents
```suggestion
    if (!missionReferent) return res.status(404).send({ ok: false, code: ERRORS.NOT_FOUND });
    if (missionReferent.length > 1) return res.status(405).send({ ok: false, code: ERRORS.OPERATION_NOT_ALLOWED });
```